web/src/lib/components/shared-components/leaflet/asset-marker-cluster.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts" context="module">
+	import { createContext } from '$lib/utils/context';
+	import L from 'leaflet';
+
+	const { get: getContext, set: setClusterContext } = createContext<() => L.MarkerClusterGroup>();
+
+	export const getClusterContext = () => {
+		return getContext()();
+	};
+</script>
+
+<script lang="ts">
+	import 'leaflet.markercluster';
+	import { onDestroy, onMount } from 'svelte';
+	import { getMapContext } from './map.svelte';
+	import { MapMarkerResponseDto, api } from '@api';
+	import { createEventDispatcher } from 'svelte';
+
+	class AssetMarker extends L.Marker {
+		marker: MapMarkerResponseDto;
+
+		constructor(marker: MapMarkerResponseDto) {
+			super([marker.lat, marker.lon], {
+				icon: new L.Icon({
+					iconUrl: api.getAssetThumbnailUrl(marker.id),
+					iconRetinaUrl: api.getAssetThumbnailUrl(marker.id),
+					iconSize: [60, 60],
+					iconAnchor: [12, 41],
+					popupAnchor: [1, -34],
+					tooltipAnchor: [16, -28],
+					shadowSize: [41, 41]
+				})
+			});
+
+			this.marker = marker;
+		}
+
+		getAssetId(): string {
+			return this.marker.id;
+		}
+	}
+
+	const dispatch = createEventDispatcher<{ view: { assets: string[] } }>();
+
+	export let markers: MapMarkerResponseDto[];
+
+	const map = getMapContext();
+
+	let cluster: L.MarkerClusterGroup;
+
+	setClusterContext(() => cluster);
```

Comment:
The cluster is added to the context, but never used

Reply:
See below